File: backend/src/database/seeders/RawMaterialSeeder.php
<?php

namespace Database\Seeders;

use Illuminate\Database\Seeder;
use App\Models\RawMaterialType;
use App\Models\RawMaterialCharacteristic;
use App\Models\IramCopperMaxResistance;
use App\Models\IramCuerdaMaxResistance;

class RawMaterialSeeder extends Seeder
{
    public function run(): void
    {
        $types = [
            'Cobre',
            'PVC',
            'Master',
            'Varios',
            'Cuerda'
        ];

        foreach ($types as $typeName) {
            RawMaterialType::firstOrCreate(['name' => $typeName]);
        }

        $cobre = RawMaterialType::where('name', 'Cobre')->first();
        $cuerda = RawMaterialType::where('name', 'Cuerda')->first();

        $cobreDiameters = [
            [0.30, 256.00],
            [0.35, 184.60],
            [0.38, 156.50],
            [0.40, 141.30],
            [0.50, 90.44],
            [0.60, 62.81],
            [0.67, 50.50],
            [0.85, 31.29],
            [1.05, 20.51],
            [1.13, 17.70],
            [1.35, 12.41],
        ];

        foreach ($cobreDiameters as $data) {
            $diameter = $data[0];
            $maxResistance = $data[1];

            $characteristic = RawMaterialCharacteristic::firstOrCreate(
                [
                    'raw_material_type_id' => $cobre->id,
                    'name' => 'Diametro',
                    'decimal_value' => $diameter
                ],
                [
                    'unit' => 'mm',
                    'text_value' => null
                ]
            );

            IramCopperMaxResistance::updateOrCreate(
                ['raw_material_characteristic_id' => $characteristic->id],
                ['max_resistance_ohm_km' => $maxResistance]
            );
        }

        $cuerdaSecciones = [
            [0.75, 26],
            [1.00, 19.5],
            [1.50, 13.3],
            [2.50, 7.98],
            [4.00, 4.95],
            [6.00, 3.3],
            [10.00, 1.91],
            [16.00, 1.21],
            [25.00, 0.78],
            [35.00, 0.554],
            [50.00, 0.386],
            [70.00, 0.272],
            [95.00, 0.206],
            [120.00, 0.161],
        ];

        foreach ($cuerdaSecciones as $data) {
            $seccion = $data[0];
            $maxResistance = $data[1];

            $characteristic = RawMaterialCharacteristic::firstOrCreate(
                [
                    'raw_material_type_id' => $cuerda->id,
                    'name' => 'Seccion',
                    'decimal_value' => $seccion
                ],
                [
                    'unit' => 'mm²',
                    'text_value' => null
                ]
            );

            IramCuerdaMaxResistance::updateOrCreate(
                ['raw_material_characteristic_id' => $characteristic->id],
                ['max_resistance_ohm_km' => $maxResistance]
            );
        }
    }
}

File: backend-dotnet/src/Ikhana.Infrastructure/Data/Seeders/RawMaterialSeeder.cs
using Ikhana.Application.Common.Interfaces;
using Ikhana.Domain.Entities;
using Ikhana.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

namespace Ikhana.Infrastructure.Data.Seeders;

public class RawMaterialSeeder : IDataSeeder
{
    private readonly AppDbContext _context;

    public int Order => 5;

    public RawMaterialSeeder(AppDbContext context)
    {
        _context = context;
    }

    public async Task SeedAsync(CancellationToken cancellationToken = default)
    {
        if (await _context.Set<RawMaterialType>().AnyAsync(cancellationToken))
            return;

        var types = new[] { "Cobre", "PVC", "Master", "Varios" };
        foreach (var name in types)
        {
            _context.Set<RawMaterialType>().Add(new RawMaterialType { Name = name });
        }

        await _context.SaveChangesAsync(cancellationToken);

        var copperType = await _context.Set<RawMaterialType>()
            .FirstAsync(t => t.Name == "Cobre", cancellationToken);

        var characteristics = new (string Name, decimal Diameter, decimal MaxResistance)[]
        {
            ("0.30", 0.30m, 256.00m),
            ("0.35", 0.35m, 188.10m),
            ("0.38", 0.38m, 159.50m),
            ("0.40", 0.40m, 144.10m),
            ("0.50", 0.50m, 92.40m),
            ("0.60", 0.60m, 63.90m),
            ("0.67", 0.67m, 51.60m),
            ("0.85", 0.85m, 32.00m),
            ("1.05", 1.05m, 21.00m),
            ("1.13", 1.13m, 18.20m),
            ("1.35", 1.35m, 12.41m),
        };

        foreach (var (name, diameter, maxResistance) in characteristics)
        {
            var characteristic = new RawMaterialCharacteristic
            {
                RawMaterialTypeId = copperType.Id,
                Name = name,
                DecimalValue = diameter,
                Unit = "mm",
            };

            _context.Set<RawMaterialCharacteristic>().Add(characteristic);
            await _context.SaveChangesAsync(cancellationToken);

            _context.Set<IramOhmMaxResistance>().Add(new IramOhmMaxResistance
            {
                RawMaterialCharacteristicId = characteristic.Id,
                MaxResistanceOhmKm = maxResistance,
            });
        }

        await _context.SaveChangesAsync(cancellationToken);
    }
}
